arangod/Aql/QuerySnippet.cpp

```diff
@@ -275,9 +278,30 @@ void QuerySnippet::serializeIntoBuilder(
         auto permuter = localExpansions.find(current);
         if (permuter != localExpansions.end()) {
           auto collectionAccessingNode = dynamic_cast<CollectionAccessingNode*>(clone);
-          TRI_ASSERT(collectionAccessingNode != nullptr);
-          // Get the `i` th shard
-          collectionAccessingNode->setUsedShard(*std::next(permuter->second.begin(), i));
+          if (collectionAccessingNode != nullptr) {
+            // we guarantee that we only have one collection here
+            TRI_ASSERT(permuter->second.size() == 1);
+            std::string const& cName = collectionAccessingNode->collection()->name();
```

Comment:
i thank you can can also do a for loop over the permuter->second entry, like you do inside the graph node.

But this should be the same as you do here right now, just saves the lookup of the collection name

Reply:
Yes, forgot to add a TODO here. Known! But thanks :)